### DISCUSSION TITLE
$router doesnt work in pinia

### DISCUSSION THREAD

**Discussion by wilsntn** [quasarframework/quasar#14030] [General - Components / Directives / etc] [answered]
How can I redirect after a successful login with pinia?

```js
login() {
  console.log(this.email, this.password);
  if (this.email && this.password != null) {
    api
      .post("/login", { email: this.email, password: this.password })
      .then((response) => {
        console.log(response.data.acess_token);
        LocalStorage.set("token", response.data.acess_token);
        // HERE
        this.$router.push({ path: "/dashboard/servicos" });
      })
      .catch((err) => console.log(err));
  }
},
```

I know **this** cant work but i have no clue how to import Router.push in quasar.. can anyone help me?

**Comment by yusufkandemir** [comment, marked as answer]
It's not `this.$router`, it's `this.router`(_no $_).

**Reply by wilsntn** [reply]
Thank you for your answer but  same error 

![image](https://user-images.githubusercontent.com/91102332/180621083-0d190633-e677-40ca-8f8e-47da86fe5661.png)

![image](https://user-images.githubusercontent.com/91102332/180621116-dd5181a2-09b0-4e4e-b5b5-3d13dc3ce853.png)

![image](https://user-images.githubusercontent.com/91102332/180621147-1d075e4c-9776-4dd2-b09c-2d66566d6d23.png)

Here is my code, am i doing something wrong? i'm a newbie :(

**Reply by yusufkandemir** [reply]
@wilsntn you didn't save the file(_`auth.js`_) 🙂 Save it then try again.
![image](https://user-images.githubusercontent.com/6266078/180621207-08c5f78b-1712-40a7-a6c7-8ae7b5f352a8.png)

**Reply by wilsntn** [reply]
@yusufkandemir i already saved and got the same error..

![image](https://user-images.githubusercontent.com/91102332/180628181-e5193c1b-7101-40d6-9506-f8e4db569663.png)

here my component

![image](https://user-images.githubusercontent.com/91102332/180628195-a8485064-7fb2-4df1-88eb-b957449b7dd4.png)

here my form setup

**Comment by wilsntn** [comment]
I solved the problem, was my project structure, it was created to work with vuex and i created another with pinia.. now works fine
